### ISSUE TITLE
UI Idea : Need a general/index/summary view

### ISSUE THREAD

**Issue by jerrydiff** [jflesch/paperwork#548]
Let me start from the begining, when you enter paperwork, you are directely in the action, you arrive in the middle of your last documents, and .. thats it. I've seen poeple start paperwork and dont understand it was a document archive tool, because it mostly looks like an explorer windows with a research function. For some poeple, its an interface for scanning that lets you see the last document you have scanned.

It feels to me that it lacks what could be named a summary, or an index, a view that would provide with a glance what actually have in store, that provide a classification and reassure the user that his documents are indeed well organised, well classified, accounted for.

Here are a few starting ideas for this view, it should help reply to this question : how much documents are in each category, what are the main categories, how much documents have failed the automatic tagging (and if there's none, say it !), how many electric bills do i have stored,  how many documents are in a single / multiple categories and so on

Users are very used to being abused by softwares, silent errors, lost datas without warning, etc. they need some kind of metric and feedback to be assured it is actually working. Sometimes, a stupid green indicator saying all stuff are ok since last time goes a long way.
I must say paperwork works honnestly quite well, I know it, but it does not advertise in anyway during the time you interract with it. A summary would help in that regard, it gives the feeling that everything is well organized

**Comment by tiramiseb** [comment]
I agree. A "home screen" with at least a documents count, list of labels, etc, would be great and reassuring.

**Comment by jflesch** [comment]
Ok then, the first step is to figure out exactly what we want to display.

Obviously, we should have the total number of documents. But what else ?
I guess some statistics regarding labels could make sense. However, which one exactly ? And which representations (histogram, pie chart, ...) ?

**Comment by tYYGH** [comment]
I would appreciate:
— a list of most-accessed (most clicked-on) documents (there is for example “Échéancier” (FR) documents that I read every month) ;
— a list of the most-used searches.

**Comment by jflesch** [comment]
Hm, interesting ideas.
Regarding the most-used searches, it's kind-of-related to #119 I think.

**Comment by jerrydiff** [comment]
I also like something dynamic with the most accessed doc / used search, maybe a few of the most recent to remind of the latest activity

I would like something more static as well, like a summary of the content of the archive with a list of tags ordered by number of associated files

**Comment by mjourdan** [comment]
I agree too. Here is a [first shot](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview.png). Feel free to comment!

**Comment by jflesch** [comment]
No document list on the left ? oO

**Comment by tiramiseb** [comment]
Same comment than jflesch. I think the document list must remain visible.

**Comment by jerrydiff** [comment]
I really like that mockup, not sure the document list is mandatory, maybe a button to jump to the complete document list would be fitting ?

**Comment by jflesch** [comment]
I think it is. Quite often I just want to access the latest document I've scanned / imported.

**Comment by tiramiseb** [comment]
For the sake of consistency, I think the document list must always be visible...

**Comment by jerrydiff** [comment]
ok, I am already blow up that this got so much positive ideas and replies, so I am all for it as well

**Comment by mjourdan** [comment]
> I think it is. Quite often I just want to access the latest document I've scanned / imported.

Maybe my mockup was not explicit enough. The "Recent" section shows thumbnails of the 3 latests documents, with an excerpt of the very latest. Why would one need the whole document list?

> For the sake of consistency, I think the document list must always be visible...

It feels to me that it would defeat the purpose of what @jerrydiff suggests: a "welcome" screen to avoid putting users in action too abruptly.

> ok, I am already blow up that this got so much positive ideas and replies, so I am all for it as well

Do not change your mind so fast! ;-)

**Comment by tYYGH** [comment]
As far as I’m concerned, I do want to see the document list on launch, even though I am interested in a welcome screen… in the right panel.

Viewing the document list tells me at a glance what are the last 10 or so documents that I imported, and which of these are from real papers (ie. on my paper stack), compared to those that came through e-mail.

**Comment by mjourdan** [comment]
[Second shot](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview.png). Thoughts?

**Comment by tiramiseb** [comment]
I prefer the second one, however the "recent document" part is not needed: it is on the left...

**Comment by tYYGH** [comment]
+1 @tiramiseb

**Comment by jerrydiff** [comment]
I must say I prefer the first one since there is room for 3 columns of labels and 3 filters instead of 2

**Comment by jerrydiff** [comment]
also the paper/page/label/filter/size snippet is very important, its probably the most important part, It really needs to fit somewhere imho

**Comment by mjourdan** [comment]
Hum. Thanks for your feedback, folks. It seems that we raised in this topic at least four issues:
1. Users are (too) directly put in action, UX is a bit harsh.
2. Some Users don't understand what paperwork is for: is it a file manager? Is it a scanner application? Is it a document reader?
3. Different users have different flows: some often access the same papers, others need to see the last imports, some just archive papers and never read them, some heavily use labels...
4. A few numbers about what is in store would be nice.

Your comments tend to show that having the file panel on startup is ok and we do not need a whole new screen for welcoming users. Please find [new mockups here](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview-pane.png).

**Comment by tYYGH** [comment]
Happy New Year!

@mjourdan I like your second Overview Pane mockup better, because it is inside the window, and the window is probably already on the left side of the screen, if not maximized. The first mockup would either have the new Pane appear partly out of view, or shift/resize the window unnecessarily.

Some questions:
* Is the intent to have this Overview Pane showing at start-up, or would it be visible only if the user clicks the top-left button? In the latter case, there is no welcome screen, is there?
* Any reason why you decided not to use the document-area after all? It is big, and empty at launch. That said, I can see how it kinds-of misuses the area for something it was not intended for, with obvious consequences if we want to show the Overview again after some usage, and the document-area is not empty any more …

Rel. “A few numbers about what is in store would be nice”, I suppose you would like some stats from some users; here are mine: 2210 documents, spanning 25 years; 57 labels, with 12 being used less than 10 times (min. 1), 17 being used more than 100 times (max. 762); 25 documents with no labels.

**Comment by jflesch** [comment]
@mjourdan : I clearly prefer your second mockup. As @tYYGH pointed out, it avoids various annoying issues relative to the window size.

Just some random idea : Maybe we could make it possible to click on the stats below (number of documents, number of pages, etc). It could then display much more detailed statistics in the 'document-area'.

BTW, just from a technical standpoint, it's possible for me to display charts (pies, bar chart, etc).

**Comment by mjourdan** [comment]
@tYYGH:
- this would be hidden on startup and visible on click. No welcome screen in this case. The overview pane aims to cover points 3 and 4 (see my message referring to the mockups)
- I have other plans to cover points 1 and 2 by filling up the document area on first launch. I guess I'll have to split to a separate ticket for that.

> I suppose you would like some stats from some users
Actually I was thinking about finding a way to display those stats inside the GUI, as discribed by @jflesch (yeah, could be nice to have some visual stuff like charts). Thank you anyway.

**Comment by mjourdan** [comment]
The welcome screen I talked about is under #563, @tYYGH.

Would be glad to read your thoughts about that on this new ticket, folks.

**Comment by jflesch** [comment]
(label 'to study' removed --> the [2nd mockup](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview-pane.png) is just what we need IMHO).

**Comment by jflesch** [comment]
I'm going to start working on this one.

However, I think I will merge a little the ideas from the 1st mockup with the 2nd one:
I will go with the 2nd mockup, but in "Recent", I will display the last 3 documents used instead of just the last one. IMHO, having just one document in "Recent" reduce too much the usefulness of this feature. The 2 others will be displayed below the first one, will be smaller, and will be on the same line. Only their thumbnail and labels will be displayed, but no text. (I will display the first one as in the 2nd mockup).

**Comment by mjourdan** [comment]
Have you heard about [libdazzle](https://people.gnome.org/~chergert/docs/libdazzle/)? Christian Hergert introduced this [here](https://blogs.gnome.org/chergert/2017/05/31/razzle-dazzle/) (there are a couple more videos on his youtube channel). It could help to make nice sliding panels.

**Comment by jflesch** [comment]
No I didn't know it. And even just the suggestion box could be handy for the search field, because my current implementation is quite buggy (most of the time, suggestions don't even appear at all ...)

**Comment by jflesch** [comment]
The documentation is sometimes very .. succinct, but still, using this library could help quite a lot.
The only problem is that it means another extra non-python dependency .. that is not packaged in Debian yet ... :/. Time to make Flatpak work I guess.

**Comment by jflesch** [comment]
I've noted the URL. I'll have a better look at it later.
In the meantime, I see nothing in libdazzle that could be useful for this ticket (or did I miss something in your mockups ?), so I'll keep working on it with just my current tools :)

**Comment by mjourdan** [comment]
I was thinking of a [push effect](https://tympanus.net/Development/SidebarTransitions/) to reveal the left pane. I think it is better to have both the filter panel and the document list side-by-side, to make it clear the later depends on the former. Otherwise, having the document list to be refreshed hidden by the filter panel could confuse user. I thought libdazzle could help for this (as well as for resizing the document pane [like this](https://www.youtube.com/watch?v=aqFYV39wAWA)).

Besides libdazzle, I didn't get how you were intending to merge the left pane with the overview mode. So I tried to make a mockup implementing a ["recent" filter in the left pane](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview-pane-3.jpeg). Does it match what you mean?

**Comment by jflesch** [comment]
Hm, I see. The push effect is an interesting idea. It could work.
And Libdazzle is not required for that. The push effect can easily be done with a GtkRevealer. And regarding the resizing video, it's actually showing how the text in the GtkTree is updated, but I don't use GtkTree and based on your mockup, I won't even have to resize the document list.
.. but I'm a little bit lost ; Is the left pane part of this ticket ? oO

By "overview mode", do you mean the summary (last documents opened, statistics, etc) ? If so, I don't really how it is related to the left pane ?

**Comment by jflesch** [comment]
I'm going to assume the left pane part of this ticket too, because I don't know of any other ticket about that.
However, I'll start by making the summary view, and then work on the left pane

**Comment by jflesch** [comment]
https://github.com/openpaperwork/paperwork/commit/d9167a5ffe3e71bc8cf44d530d5288956932e186 : Keep track of the last documents and the last searches made

**Comment by jflesch** [comment]
1686db5dd2c68615d2047554882c17dc74878e1c : switch between summary view / document content view (summary view is pretty much empty for now)

**Comment by mjourdan** [comment]
By "overview mode", I was thinking about one of those [landing pages](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview.png), which I think you were talking about when speaking of "I will display the last 3 documents used instead of just the last one."

But as you stated on 23 april, to me it was clear that the relevant proposal to solve this issue was the overview pane. Reading your message from yesterday, I was very surprise to read you were thinking of implementing some mix of the twos.

In my opinion, the overview/landing page mentioned above breaks the UX, ans is totally obsoleted by a separate set of features:
- overview pane (see this ticket)
- onboarding page (see #563)

Sorry if things got confused here. The "experiments" folder in my repo is for getting feedback only, it's a bit risky to start implementing stuff landing there. Anyway, I'll take a look at your instable branch to see what you had in mind, perhaps it works better than what I draw.

**Comment by jflesch** [comment]
For me, the [left pane](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview-pane-3.jpeg) has a goal entirely different from the [summary](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview.png) (landing page number 2).

The left pane is complementary of the search dialog. Except it's not about building one search, but about managing set of searches.

The summary is .. well, a summary of what you have in store. It's about getting statistics and a bird-eye view.  I assume you mean it breaks the UI because this space is usually used to display document content, right ? But even currently, when you start Paperwork, you have a basic welcome screen (Paperwork logo + version).

So actually, having both could make sense to me.

Or we could put everything in the left pane (last searches, last documents, saved searched, etc). I guess it could make sense too.

**Comment by mjourdan** [comment]
> For me, the left pane has a goal entirely different from the summary (landing page number 2).

Hmm. On the contrary, I see a bit of overlapping. As stated on a previous comment, we can find in this ticket at least [4 different issues](https://github.com/openpaperwork/paperwork/issues/548#issuecomment-270019114).  IMHO, the left pane does provide numbers in a bird's-eye view, to see what's currently in store. The summary in that regard doesn't provide more value (if you compare the mockups, you'll see there is almost equal info in the left pane and in the overview). It breaks the UI because it appears in place of the content but doesn't behave like actual documents. Moreover, the excerpt of the recent papers would probably be better replaced with fixing #623 for the whole document list.

Sure, we could imagine having deeper statistics, for example giving the progression on the past few months and making projections on how much space will be consumed. But that would deserve a whole separate ticket I think.

> Or we could put everything in the left pane (last searches, last documents, saved searched, etc). I guess it could make sense too.

That's already pretty much the case, a part from recent (which is worth adding and already figures on [some of my sketches](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments/overview-pane-3.jpeg)) and last searches (which are not that useful, are they?). Though that 3 columns layout needs proper wireframes (I'm working on it).

**Comment by mjourdan** [comment]
I made the wireframes I talked about to deal with this [3 columns layout](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/3-columns-layout-switching.png). Thoughts welcome, folks.

**Comment by jflesch** [comment]
I'm starting to think you're right and this summary is pretty much made useless by the 3rd column.

Regarding the 3 columns layout, I'm wondering: Instead of a 3rd columns, why not just switch the content of the columns like done currently with document properties ? It would use much less space on the screen.
(I may have already asked that, but I don't remember the answer)

**Comment by mjourdan** [comment]
> Regarding the 3 columns layout, I'm wondering: Instead of a 3rd columns, why not just switch the content of the columns like done currently with document properties ? It would use much less space on the screen.

This is indeed an option. I would expect the following benefits:

1. Having both the filters column and the document list visible should make more obvious how they are related.
2. On big hi-def screens, using the space actualy available would allow a convenient and very fast way to apply filters.
3. Biggest reason is everything would be visible on startup; user would both have an overview of what's in store, what are the most recent documents, and could read the latest one

On the other hand:

1. Indeed, on small screens, a new column would eat space needed for more valuable content. That's why I tried to design so the software would remain perfectly usable with that column hidden.
2. Maybe a "previous" button would be easier to use than a dropdown to change the layout.

Before implementing anything, I think I should make some prototypes for user testing to know which works best. :/

> (I may have already asked that, but I don't remember the answer)

I don't think you did, or sorry if I missed something.

**Comment by jflesch** [comment]
> Before implementing anything, I think I should make some prototypes for user testing to know which works best. :/

Prototypes or mockups ?

> I don't think you did, or sorry if I missed something.

Nah, don't worry, I have a pretty crappy memory at times, so I probably never asked in a first place. (sometimes I'm surprised I haven't driven my girlfriend mad yet ... :)

**Comment by mjourdan** [comment]
> Prototypes or mockups ?

Prototypes, that is to say interactive mockups. I would like to give a try to [gravit designer](https://designer.io/#download) (this is nonfree software :-( )


> Nah, don't worry, I have a pretty crappy memory at times, so I probably never asked in a first place. 
> (sometimes I'm surprised I haven't driven my girlfriend mad yet ... :)

Maybe you did, it's just you don't remember :-p

**Comment by mjourdan** [comment]
I gave a try to Gravit Designer for an other project, and prefer to keep Inkscape for Paperwork. I am not sure Gravit actually features prototyping anyway.

I made some more iterations on the idea of an overview pane. Here is [what it could look like](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/filters-overview-pane.png). Thoughts?

**Comment by jflesch** [comment]
1) Search dialog: looks cool :)

2) Other mockups: I'm really not fond of the idea of having 2 left-side columns at any point. Even less fond of having one pushing the rest of the window content. As stated before, I still think this is going to be messy to manage: When displaying the extra column, do we extend the window to the right ? If we don't, what happens if the window was too small ?
In other words, I think sliding things may end up being a problem.

My 2 cents:
Both columns are about search. So it would make sense to display both at the very same place and switch them as we do currently with the document list and the document properties. When paperwork starts, it display first the overview column. In the overview, we make everything clickable (for instance the label names) --> one click and it switches to the search result list.

Regarding large screen, there is another way we could use them: We could display the document properties to the right side of the window (I think I remember seeing that in one of your other mockups ?). I think it would nicely enforce the idea "left = document listing ; right = current document", no ?

**Comment by mjourdan** [comment]
Thanks for discussing this!

> As stated before, I still think this is going to be messy to manage: When displaying the extra column, do we extend the window to the right ? If we don't, what happens if the window was too small ?
In other words, I think sliding things may end up being a problem.

The size of the column wouldn't change, because the content of the document viewer would also slide. This would eventually make one part of the page being out of bounds, as (not) shown on the mockups.

> Both columns are about search.

Not quite. In my mind, each column shows the children of the previous one:
- 1st (new) shows high-level abstractions (labels, saved searches, numbers…)
  - 2nd (already existing) shows a list of documents
    - 3rd (already existing) shows a list of pages
      - 4th (moved) shows specific details (the metadata)

To me it seems natural to show simultaneously parents and children side-by-side, just like we already do with the documents list and the reader.

> So it would make sense to display both at the very same place and switch them as we do currently with the document list and the document properties. When paperwork starts, it display first the overview column. In the overview, we make everything clickable (for instance the label names) --> one click and it switches to the search result list.

Ok, I'll do some mockups for that!

> Regarding large screen, there is another way we could use them: We could display the document properties to the right side of the window (I think I remember seeing that in one of your other mockups ?). I think it would nicely enforce the idea "left = document listing ; right = current document", no ?

[There](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/edit-information.png)! Yep, that would be a nice move disregarding the overview issue. Though this would be of little benefit for large screens, as this is not something useful enough for keeping it open permanently, even less simultaneously with the documents list.

**Comment by mjourdan** [comment]
I uploaded new mockups to presentator.io, in the hope we could get a feeling of what works best. You'll see on hover you can click on certain zones and discover what happens then. Sadly, this is no true prototype solution.

Nevertheless, it's here: https://app.presentator.io/fr/APieffK2?v=2&s=1&m=preview (do not forget to try both variant 3 and variant 4!)

As usual, you know you can speak frankly! Do tell me what you would like to see if you want to go deeper into variant 4.

**Comment by mjourdan** [comment]
I updated the [overview pane mockups](https://github.com/mjourdan/gnome-design/blob/master/paperwork/experiments-2.0/filters-overview-pane.png) with new variants. I think numbers 6 and 7 are worth taking a look at, because they basically keep the fix-width 2 panels layout, while keeping context and the most interesting stuff visible.

**Comment by jflesch** [comment]
The variant #4 feels clearly cleaner to me.

**Comment by mjourdan** [comment]
Really it's the one I like the least:

- it shows either the list of tags, or the list of documents, but not both at the same time
- the document shown on the left side is somewhat out of context, not related in any way with the list of tags
- not sure how well it would work if we had the app menu icon in the headerbar, before the back button
- we could not merge the app menu with the left pane, which is a pretty common pattern (in web and material design, not in gnome though)

So maybe this looks cleaner, but I feel like this would fail to give an "overview" and would be really the less usable of all the experiments here.

@jerrydiff , @tiramiseb , @tYYGH , maybe you have an opinion on this?

**Comment by tYYGH** [comment]
As I see things, this new panel is interesting, but it is not part of a “left-to-right” relationship, if you see what I mean. For example, the “right” document panel is linked to the “middle” one: the page displayed on the right is part of the document that is selected in the list; but the “middle” list is not linked to the “left” (new) panel: there is no concept of _current_ element in the new “left” pane.
In my eye, this rules out variants 1 to 3.

Although I do not like variant 4 much, for the reasons given by @mjourdan, it does have some merit: the _current_ state of the “left” panel becomes implicit in the fact that we kind-of entered an item in that panel, much like we enter a directory in Nautilus. This is symbolized by the left-pointing arrow at the corner of the window, which indicates that the displayed list is in accordance to the item that was selected on the parent pane (although there is no indication of which that is), much like the details of a document, in the details panel, are in accordance to the document that was clicked on, on the previous panel (the list).
In short, it is at least somewhat consistent.

This leaves variants 5 to 7, which I like better. However, variants 6 and 7 seem to imply the kind of relationship already found in variants 1 to 3, which is IMHO non-existent.

My all-favourite is thus variant 5.

From my point of view, this new pane is nothing more than a “new-age” menu. Sure, it is dynamic, but that is nothing new (numerous software give access to the list of recent documents, for example).
For this reason, I would ask one question (all the more since there are actual menu items in this panel): why not make this panel ***the*** Gnome menu?

**Comment by jflesch** [comment]
Actually, when I think of it, I'm not sure I see a real difference between #4 and #7 ? oO

**Comment by tiramiseb** [comment]
I prefer #5 too.
Maybe #7, but there is no advantage in pushing the document list behind the document reader...

Regarding making it "the" Gnome menu, I'm not sure I understand which menu you are talking about. If you are talking about the menu displayed when you click on the window name in the topbar, I'm against it.

It could definitely be a regular menu, with dynamic entries (but I don't know if we can put label icons, colors, etc, in a menu...)

**Comment by tiramiseb** [comment]
jflesh: are you talking about #5 vs #7, or really #4 vs #7 ?
Because the difference between #4 and #7 is obvious to me...
The difference between #5 and #7 is just about the behaviour of the document list.

**Comment by tYYGH** [comment]
@tiramiseb Yes, that’s the menu I was referring to: the unigue application menu in the black Gnome3 top-bar.

**Comment by tiramiseb** [comment]
As I understand it, this menu is for occasional actions (parameters, etc), not for day-to-day use.

**Comment by tYYGH** [comment]
@jflesch :
4: the new panel and the current left panel alternate: it is one or the other, at the exact same location.
7: the new panel is hidden on the left, and slides out from the left, pushing the current left panel 2/3 of its width to the right, where it slides under the right panel… more or less…

**Comment by jflesch** [comment]
Ok, because if I have to implement #4, there would still be sliding/pushing (as currently in 1.2.4). So the difference is actually that it would stop at 2/3 or 3/4 of the way, right ?

**Comment by mjourdan** [comment]
Thanks all for your comments! Below answers to specific points.

> but the “middle” list is not linked to the “left” (new) panel: there is no concept of current element in the new “left” pane.

There is, actually, that's why there is a light blue bar and small police changes (the very same treatment used between documents list and document reader). On startup it always shows "all papers", but it is intended to apply filters as well.

> why not make this panel the Gnome menu?

That's precisely what I did with all variants except number 4!? This could not be done on number 4 because the app menu has to be behind an always visible button showing the app icon (so we can't use the back button for that). Or did I miss your point?

> Maybe #7, but there is no advantage in pushing the document list behind the document reader...

I think there is, as it shows thumbnails of recent documents on startup, and thumbnails of the documents matching the selected filter. Variant number 5 though shows a bit of the document list, but an absolutely useless part of it.

> I wasn't sure if the mockups show things in-motion or once they are at the end of each motion :/

@tYYGH explained well. Mockups do not show transitions and show the end of each motion!

As it appears that the mockups are not as obvious as I thought (shame on me!), I turned them into wireframes to demonstrate the same scenario: Start Paperwork -> Apply a filter -> hide the overview column.

Feel free to comment again or suggest totally different approaches if you like.

**Comment by jflesch** [comment]
Ok, I eventually got the difference between 4 and 7 .... :)

**Comment by jflesch** [comment]
I like 7 better. But here are some things I would prefer to discuss:
* In the current implementation, the assumption is that the users more often use the last documents scanned than the other. This is why the default is the whole document list, ordered by reversed date.
* When the user will start Paperwork for the first time(s), the overview will be almost empty. Is that what we want to show as soon as it start ?
* A 2/3 split will be *hard* to implement and really bug-prone.

My questions:
* Do we need the overview immediately when we start ? Wouldn't it be better to just display it when the user request it (application menu for instance) ?
* Do we need a 2/3 split ? The animation should show clearly enough what happens, and entirely replacing the left pane (as currently for the doc proprerties) would give more room for the overview, no ?

**Comment by mjourdan** [comment]
> In the current implementation, the assumption is that the users more often use the last documents scanned than the other. This is why the default is the whole document list, ordered by reversed date.

That would stay true.

> When the user will start Paperwork for the first time(s), the overview will be almost empty. Is that what we want to show as soon as it start ?

So would be the document list. The list of tags may even increase faster than the document list, for users who set a couple of labels per document.

> Do we need the overview immediately when we start ? Wouldn't it be better to just display it when the user request it (application menu for instance) ?

I think showing it on startup would better answer the different problems reported by @jerrydiff (see the list [here](https://github.com/openpaperwork/paperwork/issues/548#issuecomment-270019114)).

> Do we need a 2/3 split ? The animation should show clearly enough what happens, and entirely replacing the left pane (as currently for the doc proprerties) would give more room for the overview, no ?

The overview would not be complete if the list of labels and the list of the most recent matching documents were not shown at the same time. So, the 2/3 thing is an idea to show a complete enough overview while keeping the document reader unchanged. This is intended mainly for small screens, as an alternative to the push effect shown in variants 3 and 9 (which I prefer).

Adding a new pane provides useful info and allows for quicker navigation, but is shows only labels and counters, so it needs less space than the document list which shows thumbnails too (and in the mockups, they also show excerpt). This is quite a common pattern nowadays, not really on GNOME but there are several examples ([e-mail clients](https://www.monterail.com/blog/2016/the-power-of-email-clients-why-did-we-redesign-thunderbird?utm_source=Thunderbird&utm_campaign=Dribbble&utm_medium=blogpost%22), note-taking apps like [ulysses](https://ulyssesapp.com/#top), [bear](http://www.bear-writer.com/), [boostnote](https://boostnote.io/), [notes-up](https://github.com/Philip-Scott/Notes-up)…)

**Comment by jflesch** [comment]
It makes sense.
However, in all the example apps you shown, what we call the overview cannot be hidden. I guess nowadays on 16/9 screens it makes sense.
Maybe as a first step, we should forget about small screens ? I'm starting to think it's making things really much more complicated and maybe it would not be that useful ?

**Comment by tYYGH** [comment]
I wish to defend the “small screen” camp! I use quotes because in truth, I’m not that much interested in small screens, but…

I **very often** need to have Paperwork opened **next to** another window (email, web, accounting…), because these documents are not an end in themselves; they are documents related to:
* bank accounts that eventually need to be reconciled,
* commercial services (phone, utilities…), the web sites of which make no effort in tracking what was already downloaded (they all wish to be your primary storage…),
* quotations submitted to you, for which you are expected to give an answer,
* documents you need to refer to, while placing an order…

It is thus very important to me, that Paperwork works in a width equal to between ½ and ¾ of the screen width, ie. roughly 640 pixels or more.

**Comment by jflesch** [comment]
@mjourdan : A while ago you mentioned a lib for various GTK animations. I can't find it back. Can you remind me what it was please ?

**Comment by mjourdan** [comment]
> I like it, with a preference for the first popup, and with a condition: Paperwork would have to remember the layout between restarts

Yeah, sure!

> What about an “horizontal accordion button/switch”?

I'm not sure I understand. Basically, you mean something like a [view switcher](https://developer.gnome.org/hig/stable/view-switchers.html.en) with icons in it showing either 3, 2 or 1 pane?

**Comment by tYYGH** [comment]
@mjourdan , not quite… It sure is not easy to describe, but it would IMHO be easy to understand.
Let me explain differently. You see, those rolled rulers several meters long?
eg: https://i2.cdscdn.com/pdt2/2/1/4/1/700x700/46214/rw/metre-ruban-3-m-x-16-mm.jpg

When only 1cm is out, you only see the 1st cm.
If you pull 1 more cm, you see the 1st cm, _and_ the 2nd cm.
If you pull 1 more cm, you see the 1st cm, and the 2nd cm, _and_ the 3rd cm.

Now imagine, when you release the ruler back into its box, the cm do not actually disappear, but instead “fold” like https://www.storepascher.com/317-large_default/store-bateau-tamisant-taffetas.jpg, so that a tiny bit remains visible to remind us of their existence.

Let “¦“ be a vertical separation in the widget, and let “⇂” be a handle for the mouse to drag, and let “|+—” be the window panels’ borders. We would have:

One area visible:
```
+——————————————————————————————————————————————
| ¦¦¦Page⇂
+——————————————————————————————————————————————
|
|   The Page area
|
```

Two areas visible:
```
+——————————————————————————————————————————————
| ¦¦List¦Page⇂
+———————————————+——————————————————————————————
|               |
| The List area | The Page area
|               |
```

Three areas visible:
```
+——————————————————————————————————————————————
| ¦Find¦List¦Page⇂
+———————————————+———————————————+——————————————
|               |               |
| The Find area | The List area | The Page area
|               |               |
```

**Comment by mjourdan** [comment]
Dragging would probably work on touchscreens but I'm less confident with mouse/touchpad. Anyway, I tried to iterate over your accordion idea, and combined this with a pathbar. Here is [the thing](https://raw.githubusercontent.com/mjourdan/gnome-design/master/paperwork/experiments-2.0/3-columns-layout-switching-2.png).
